### PR TITLE
Add FindReferenceUsagesHandler

### PR DESCRIPTION
**Pull request by nosami**
fixes VSTS #584824 (#4194)

### CODE REVIEW

**Comment by nosami** [comment]
findreferenceusages

**Comment by Therzok** [comment]
I'd go about this a bit differently:
a) provide an extension point that a common command handler in MonoDevelop.Refactoring can use
b) make the current C# command handler call the extension nodes

**Comment by nosami** [comment]
@Therzok: Yes, ok. I'll refactor.

File: main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring/RefactoringService.cs [hunk 1]
```diff
@@ -270,6 +283,19 @@ public static MonoDevelop.Ide.Editor.DocumentLocation GetCorrectResolveLocation
 			}
 		}
 
+		public static async Task FindReferenceUsagesAsync(Projects.ProjectReference projectReference)
+		{
+			var monitor = IdeApp.Workbench.ProgressMonitors.GetSearchProgressMonitor (true, true);
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring/RefactoringService.cs]: Nit:
```
using (var monitor = ...) {

}
```

**Pull request merged**